### ISSUE TITLE
'MongoError, topology was destroyed' when no Internet connection

### ISSUE THREAD

**Issue by Dattaya** [Automattic/mongoose#4498]
I see this error each time an Internet connection is absent.
My software:
```bash
>ver

Microsoft Windows [Version 10.0.14393]

>npm ls mongoose
`-- mongoose@4.5.9

>node -v
v6.3.1
```

Stacktrace:

```bash
MongoError: topology was destroyed
    at D:\Project\node_modules\mongodb\lib\db.js:833:78
    at executeCommand (D:\Project\node_modules\mongodb\lib\db.js:287:67)
    at Db.command (D:\Project\node_modules\mongodb\lib\db.js:329:44)
    at Db.dropDatabase (D:\Project\node_modules\mongodb\lib\db.js:831:49)
    at mongoose.connect (D:\Project\server\api\index.js:14:28)
    at D:\Project\node_modules\mongoose\lib\connection.js:284:19
    at NativeConnection.Connection.error (D:\Project\node_modules\mongoose\lib\connection.js:449:12)
    at D:\Project\node_modules\mongoose\lib\connection.js:480:15
    at D:\Project\node_modules\mongoose\lib\drivers\node-mongodb-native\connection.js:69:21
    at D:\Project\node_modules\mongodb\lib\db.js:231:14
    at .<anonymous> (D:\Project\node_modules\mongodb\lib\server.js:240:9)
    at g (events.js:286:16)
    at emitTwo (events.js:106:13)
    at emit (events.js:191:7)
    at .<anonymous> (D:\Project\node_modules\mongodb-core\lib\topologies\server.js:218:12)
    at g (events.js:286:16)
    at emitTwo (events.js:106:13)
    at emit (events.js:191:7)
    at .<anonymous> (D:\Project\node_modules\mongodb-core\lib\connection\pool.js:119:12)
    at g (events.js:286:16)
    at emitTwo (events.js:106:13)
    at emit (events.js:191:7) name: 'MongoError', message: 'topology was destroyed' }
Info: No mac address - uniqid() falls back to uniqid.process(). no interfaces found
```

How to reproduce: turn off internet connection, start a project that was working before, you should see this error at least on Windows.
I don't know what's causing it—upstream repo or `mongoose`.

**Comment by vkarpov15** [comment]
Is the project running when you turn off the internet, or is does this error happen when mongoose is initially trying to connect?

**Comment by Dattaya** [comment]
> Is the project running when you turn off the internet, or is does this error happen when mongoose is initially trying to connect?

When I start the server.

`127.0.0.1` instead of `localhost` fixed this problem (`'mongodb://127.0.0.1:27017/...'`); don't know why though.

**Comment by Mann05** [comment]
Please also check mongo is running or not ?
`mongoose.connect(uri, 
    { server: { 
        // sets how many times to try reconnecting
        reconnectTries: Number.MAX_VALUE,
        // sets the delay between every retry (milliseconds)
        reconnectInterval: 1000 
        } 
    }
);`
